metadata/beam_configuration.xml

```diff
@@ -4,27 +4,98 @@
+                <type>string</type>
+                <mandatory-flag>false</mandatory-flag>
+                <externally-managed-flag>false</externally-managed-flag>
+            </attribute-definition>
+            <attribute-definition attribute-id="beamStoreId">
+                <display-name xml:lang="x-default">Beam Store ID</display-name>
+                <description xml:lang="x-default">ID of the store associated with the beam.</description>
+                <type>int</type>
+                <mandatory-flag>false</mandatory-flag>
+                <externally-managed-flag>false</externally-managed-flag>
+            </attribute-definition>
+        </custom-attribute-definitions>
+        <group-definitions>
+            <attribute-group group-id="Beam">
+                <display-name xml:lang="x-default">Beam</display-name>
+                <attribute attribute-id="beamNonprofitId"/>
+                <attribute attribute-id="beamSelectionId"/>
+                <attribute attribute-id="beamCartId"/>
+                <attribute attribute-id="beamSFCartId"/>
+                <attribute attribute-id="beamStoreId"/>
+            </attribute-group>
+        </group-definitions>
+    </type-extension>
+    <type-extension type-id="Basket">
+        <custom-attribute-definitions>
+            <attribute-definition attribute-id="beamNonprofitId">
+                <display-name xml:lang="x-default">Beam Nonprofit ID</display-name>
+                <description xml:lang="x-default">The ID of the nonprofit associated with the beam.</description>
+                <type>int</type>
+                <mandatory-flag>false</mandatory-flag>
+                <externally-managed-flag>false</externally-managed-flag>
+            </attribute-definition>
+            <attribute-definition attribute-id="beamSelectionId">
+                <display-name xml:lang="x-default">Beam Selection ID</display-name>
+                <description xml:lang="x-default">ID of the selection in the beam.</description>
+                <type>string</type>
+                <mandatory-flag>false</mandatory-flag>
+                <externally-managed-flag>false</externally-managed-flag>
+            </attribute-definition>
+            <attribute-definition attribute-id="beamCartId">
+                <display-name xml:lang="x-default">Beam Cart ID</display-name>
+                <description xml:lang="x-default">ID of the cart in the beam.</description>
+                <type>string</type>
+                <mandatory-flag>false</mandatory-flag>
+                <externally-managed-flag>false</externally-managed-flag>
+            </attribute-definition>
+            <attribute-definition attribute-id="beamSFCartId">
+                <display-name xml:lang="x-default">Beam SF Cart ID</display-name>
+                <description xml:lang="x-default">The Salesforce Cart ID for the beam.</description>
```

Comment:
`request` The descriptions here ending with `the beam.` seem like a bit of odd English. Perhaps we can look at these again?